larva/src/main/java/nl/nn/adapterframework/testtool/TestTool.java

```diff
@@ -4203,138 +4205,123 @@ public static String replaceRegularExpression(String string, String from, String
 									parts.add(stringPart);
 								}
 								j++;
 							}
 						}
-						Part allParts[] = new Part[parts.size()];
+						Part[] allParts = new Part[parts.size()];
 						allParts = parts.toArray(allParts);
-						MultipartRequestEntity multipartRequestEntity = new MultipartRequestEntity(
-								allParts, new PostMethod().getParams());
+						MultipartRequestEntity multipartRequestEntity = new MultipartRequestEntity(allParts, new PostMethod().getParams());
 						ByteArrayOutputStream requestContent = new ByteArrayOutputStream();
 						try {
 							multipartRequestEntity.writeRequest(requestContent);
 						} catch (IOException e) {
-							errorMessage(
-									"Could not create multipart: " + e.getMessage(),
-									e, writers);
+							errorMessage("Could not create multipart: " + e.getMessage(), e, writers);
 						}
 						request.setContent(requestContent.toByteArray());
-						request.setContentType(
-								multipartRequestEntity.getContentType());
+						request.setContentType(multipartRequestEntity.getContentType());
 						value = request;
 					}
 					else{
-						MockHttpServletRequest request = new MockHttpServletRequest();
-						value = request;
+						value = new MockHttpServletRequest();
 					}
 				} else {
-					value = properties.getProperty(property + ".param" + i + ".value");
+					value = properties.getProperty(property + _param + i + ".value");
 					if (value == null) {
-						String filename = properties.getProperty(property + ".param" + i + ".valuefile.absolutepath");
+						String filename = properties.getProperty(property + _param + i + ".valuefile.absolutepath");
 						if (filename != null) {
-							value = readFile(filename, writers);
+							value = new FileMessage(new File(filename));
 						} else {
-							String inputStreamFilename = properties.getProperty(property + ".param" + i + ".valuefileinputstream.absolutepath");
+							String inputStreamFilename = properties.getProperty(property + _param + i + ".valuefileinputstream.absolutepath");
 							if (inputStreamFilename != null) {
-								try {
-									value = new FileInputStream(inputStreamFilename);
-								} catch(FileNotFoundException e) {
-									errorMessage("Could not read file '" + inputStreamFilename + "': " + e.getMessage(), e, writers);
-								}
+								errorMessage("valuefileinputstream is no longer supported use valuefile instead", writers);
 							}
 						}
 					}
 				}
-				if (value != null && value instanceof String) {
-					if ("node".equals(properties.getProperty(property + ".param" + i + ".type"))) {
+				if(typesRequireStringType.contains(type) && value instanceof Message) {
+					try {
+						value = ((Message)value).asString();
+					} catch (IOException e) {
+						errorMessage("Could not convert parameter '" + name + "' with value: " + value +" to string", e, writers);
+					}
+				}
+				if (value instanceof String) {
```

Comment:
Instead of introducing `typesRequireStringType` you could just do `value.asString()` instead of `(String)value`